src/smpi/smpi_replay.cpp

```diff
@@ -942,18 +1105,70 @@ static void action_allToAllv(const char *const *action) {
   xbt_free(recvdisps);
 }
 
-void smpi_replay_run(int *argc, char***argv){
+
+/*In previous versions, we called smpi_send_process_data (implemented on
+ * smpi_base.c). The problem is that that function needs to have at least one
+ * process in the destination host. This is not always true when using a
+ * centralized LB approach.*/
+void smpi_replay_send_process_data(double data_size, sg_host_t host)
+{
+  smx_activity_t action;
+  sg_host_t host_list[2];
+  double comp_amount[2];
+  double comm_amount[4];
+  
+  TRACE_smpi_send_process_data_in(smpi_process_index());
+
+  host_list[0] = SIMIX_host_self();
+  host_list[1] = host;
+  comp_amount[0] = 0;
+  comp_amount[1] = 0;
+  comm_amount[0] = data_size;
+  comm_amount[1] = 0;
+  comm_amount[2] = 0;
+  comm_amount[3] = 0;
+  action = simcall_execution_parallel_start("data_migration", 2, host_list, 
+	    comp_amount, comm_amount, 1.0, -1.0, 0);
+  simcall_execution_wait(action);
+
+  TRACE_smpi_send_process_data_out(smpi_process_index());
+}
+
+
+/* Based on MSG_process_migrate [src/msg/msg_process.c] */
```

Comment:
What do you suggest here, then?
Is this really a duplication? I don't think so.